alfred-telemetry-solr/alfred-telemetry-solr6/src/main/java/eu/xenit/alfred/telemetry/solr/monitoring/binder/SolrSnapshootMetrics.java

```diff
@@ -4,95 +4,92 @@
 import io.micrometer.core.instrument.MeterRegistry;
 import io.micrometer.core.instrument.Tags;
 import io.micrometer.core.instrument.binder.MeterBinder;
-import java.lang.reflect.Field;
-import java.text.ParseException;
-import java.text.SimpleDateFormat;
-import java.util.Map;
-import java.util.TimeZone;
 import org.alfresco.solr.AlfrescoCoreAdminHandler;
+import org.apache.solr.common.params.MapSolrParams;
+import org.apache.solr.common.params.SolrParams;
 import org.apache.solr.common.util.NamedList;
 import org.apache.solr.core.SolrCore;
 import org.apache.solr.handler.ReplicationHandler;
+import org.apache.solr.handler.RequestHandlerBase;
+import org.apache.solr.request.LocalSolrQueryRequest;
 import org.apache.solr.request.SolrRequestHandler;
+import org.apache.solr.response.SolrQueryResponse;
 import org.slf4j.Logger;
 import org.slf4j.LoggerFactory;
 
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Map;
+import java.util.TimeZone;
+
 public class SolrSnapshootMetrics implements MeterBinder {
+    public static final String CMD_DETAILS = "details";
+    static final String COMMAND = "command";
+    public static final String CMD_BACKUP = "backup";
 
     private AlfrescoCoreAdminHandler coreAdminHandler;
     private MeterRegistry registry;
 
     private static final Logger logger = LoggerFactory.getLogger(SolrSnapshootMetrics.class);
-    private static final Map<String,String> statusMapping = Map.of(
-            "success", "1",
-            "failed", "0"
-    );
+    private static final Map<String, Long> statusMapping = Map.of("success", 1L, "failed", 0L);
     private SimpleDateFormat formatter = new SimpleDateFormat("EEE MMM dd HH:mm:ss 'UTC' yyyy");
 
     public SolrSnapshootMetrics(AlfrescoCoreAdminHandler coreAdminHandler) {
         this.coreAdminHandler = coreAdminHandler;
         formatter.setTimeZone(TimeZone.getTimeZone("UTC"));
     }
 
-    private void registerSnapshootMetrics() {
+    private void registerSnapshootMetrics() throws Exception {
```

Comment:
Don't just throw a generic Exception. 
https://stackify.com/common-mistakes-handling-java-exception/

Reply:
So this boils down to Solr code throwing Exception